toupee/data.py

```diff
@@ -205,6 +208,29 @@ def resample(self, sample_size:int=None, weights=None, replace:bool=True):
         self.resample_replace = replace
         self.__class__ = ResamplingDataset
         return self
+    
+    def copy_with_new_data(self, new_data):
+        """ Create a new copy but with new data. Useful for distillation and post-processing. """
+        new_dataset = copy.copy(self)
```

Comment:
Is a shallow copy enough? Things like numpy arrays will only be copied by reference

Reply:
the only reason why a shallow copy is enough is because we are replacing the data, so the numpy arrays don't matter